sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -251,6 +251,34 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
 
   bool AmqpValue::operator==(AmqpValue const& that) const
   {
+    if (IsNull() && that.IsNull())
+    {
+      return true;
+    }
+    if (IsNull() || that.IsNull())
+    {
+      return false;
+    }
+    if (GetType() != that.GetType())
+    {
+      return false;
+    }
```

Comment:
This logic is quite subtle and could benefit from a comment explaining the intent.

Reply:
You're right, adding a comment makes sense to me.

Off topic comment: The comments in the code were actually created by CoPilot - I typed in // and darned if the silly thing didn't get it right :)!!!.